### PR TITLE
GateOne: init at 1.2

### PR DESCRIPTION
**Pull request by tomberek**
Plenty of configuration will be needed in the settings directory.

### CODE REVIEW

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -3392,6 +3392,24 @@ let
...
+  gateone = buildPythonPackage rec {
+    name = "gateone-1.2-0d57c3";
+    src = pkgs.fetchFromGitHub {
+      rev = "11ed97c663b3e8c1b8eba473b5cf8362b10d57c3";
+      owner= "liftoff";
+      repo = "GateOne";
+      sha256 ="0zp9vfs6sqbx4d0g45kkjinfmsl9zqwa6bhp3xd81wx3ph9yr1hq";
+    };
+    buildInputs = with pkgs.pythonPackages; [tornado futures html5lib readline];
+    propagatedBuildInputs = with pkgs.pythonPackages; [tornado futures html5lib readline pkgs.openssl];
```
Comment (nckx) [inline pkgs/top-level/python-packages.nix]: You don't need to repeat `propagatedBuildInputs` in `buildInputs`.

File: pkgs/top-level/python-packages.nix [hunk 2]
```diff
@@ -3392,6 +3392,24 @@ let
...
+    meta = {
+      homepage = https://liftoffsoftware.com/;
+      description = "GateOne is a featured terminal emulator";
```
Comment (nckx) [inline pkgs/top-level/python-packages.nix]: I've never heard "featured" used like that, and it *is* far more than a terminal emulator. [What about](http://nixos.org/nixpkgs/manual/#sec-standard-meta-attributes): `"A web based terminal emulator and SSH client"`?

File: pkgs/top-level/python-packages.nix [hunk 3]
```diff
@@ -3392,6 +3392,24 @@ let
     };
   };
 
+  gateone = buildPythonPackage rec {
```
Comment (nckx) [inline pkgs/top-level/python-packages.nix]: I don't have the CPU cycles to test this (nor does Travis, apparently), but the odds of this package needing something like `disabled = ! isPy3k` (random example!) is probably high.

File: nixos/modules/services/networking/gateone.nix [hunk 4]
```diff
@@ -0,0 +1,63 @@
+{ config, lib, pkgs, ...}:
+with lib;
+let
+  cfg = config.services.gateone;
+in
+{
+options = {
+    services.gateone = {
+      enable = mkOption {
```
Comment (jagajaga) [inline nixos/modules/services/networking/gateone.nix]: Use `mkEnableOption` please.

**Comment by jagajaga** [comment]
Please don't forget to rename commit's name following our guide in `CONTRIBUTING.md`

**Comment by tomberek** [comment]
@jagajaga Changes made.

**Comment by jagajaga** [comment]
Squash commits and name the resulting commit properly.

**Comment by tomberek** [comment]
@jagajaga: Cleaned up commit.

**Pull request merged**